src/game/path.service.js

```diff
@@ -58,28 +58,39 @@ angular.module('ngGo.Game.Path.Service', [
       return;
     }
 
+    //Decrement move
+    this.move--;
+
     //Delete path choice
-    if (this.path[this.move]) {
+    if (typeof this.path[this.move] !== 'undefined') {
+      var branch = this.path[this.move];
       delete this.path[this.move];
-      this.branches--;
+      if (branch > 0) {
+        this.branches--;
+      }
     }
 
-    //Decrement move
-    this.move--;
     return this;
```

Comment:
The reason that `this.move--` should be done first is because we did the following in `advance` function:
```
    this.path[this.move] = i;

    //Increment move
    this.move++;
```
If we don't decrement it first, `this.path[this.move]` would always be `undefined`.